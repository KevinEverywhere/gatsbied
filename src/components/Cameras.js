import React from 'react'

class Cameras extends React.PureComponent {
  constructor(props){
    super();
    this.state={cameras:null}
  }
  createdAnimations(fromAnimArray, sequential){
    let myAnimations;
    if(!sequential){
      myAnimations = fromAnimArray.map((anim, i) => {
        let str = ` animation__${i}=`
        for(var z in this){
          str += `${z}: ${this[z]};`
        }
        return str.substr(0,str.length-1)
      });
    }
    return myAnimations;
  }
  componentDidMount(){
    import('aframe')
      .then(() => {
        import('aframe-extras')
          .then(() => {
            this.setState({
              cameras:(
                <a-entity id="rig" position="0 10 -3" rotation="0 0 0">
                  <a-camera far="20000" position="0 2 0" id="main-camera">
                    <a-cursor></a-cursor>
                  </a-camera>
                </a-entity>
              )})
          })
      })
      .catch((error) => console.error(error));
    }
  render() {
    return this.state.cameras;
  }
}

export default Cameras
