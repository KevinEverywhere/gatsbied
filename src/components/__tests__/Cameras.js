import React from "react"
import renderer from "react-test-renderer"
import Cameras from "../Cameras"

describe("Cameras", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Cameras />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
