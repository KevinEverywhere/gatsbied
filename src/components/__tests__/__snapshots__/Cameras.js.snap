// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cameras renders correctly 1`] = `null`;
